### PR TITLE
Move access to .getTypeRegistry after .compile.

### PR DESCRIPTION
**Pull request by rkirov**
In preparation for using options.setAssumeForwardDeclations(true),
one has to make sure that compiler.getTypeRegistry is never called
before compiler.compile(...,...,options). Otherwise, an internal
state is incorrectly built that uses the wrong value for
forwardDeclaredTypes.

### CODE REVIEW

**Comment by rkirov** [comment]
Alternatively there is `compile.init`, but at this point all the APIs look equally unsafe to me :/

File: src/main/java/com/google/javascript/clutz/DeclarationGenerator.java [hunk 1]
```diff
@@ -236,8 +236,8 @@
 
   private static final String MODULE_PREFIX = "module$exports$";
 
-  private final JSType UNKNOWN_TYPE;
-  private final JSType NUMBER_TYPE;
+  private JSType UNKNOWN_TYPE;
```
Comment (mprobst) [inline src/main/java/com/google/javascript/clutz/DeclarationGenerator.java]: while you're touching this, these should be `lowerCamelCase`

File: src/main/java/com/google/javascript/clutz/DeclarationGenerator.java [hunk 2]
```diff
@@ -302,8 +302,9 @@ public static void main(String[] args) {
         new ClutzErrorManager(
             System.err, ErrorFormat.MULTILINE.toFormatter(compiler, true), opts.debug);
     compiler.setErrorManager(errorManager);
-    UNKNOWN_TYPE = compiler.getTypeRegistry().getNativeType(JSTypeNative.UNKNOWN_TYPE);
-    NUMBER_TYPE = compiler.getTypeRegistry().getNativeType(JSTypeNative.NUMBER_TYPE);
+    // Calling compiler.getTypeRegistry() is not safe to call here,
```
Comment (mprobst) [inline src/main/java/com/google/javascript/clutz/DeclarationGenerator.java]: nit: Calling ... to call

File: src/main/java/com/google/javascript/clutz/DeclarationGenerator.java [hunk 3]
```diff
@@ -302,8 +302,9 @@ public static void main(String[] args) {
...
+    // because it initialized some internal compiler structures before
```
Comment (mprobst) [inline src/main/java/com/google/javascript/clutz/DeclarationGenerator.java]: initialize_s_

File: src/main/java/com/google/javascript/clutz/DeclarationGenerator.java (hunk 1 above)
Comment (rkirov) [inline src/main/java/com/google/javascript/clutz/DeclarationGenerator.java]: done

File: src/main/java/com/google/javascript/clutz/DeclarationGenerator.java (hunk 2 above)
Comment (rkirov) [inline src/main/java/com/google/javascript/clutz/DeclarationGenerator.java]: done

File: src/main/java/com/google/javascript/clutz/DeclarationGenerator.java (hunk 3 above)
Comment (rkirov) [inline src/main/java/com/google/javascript/clutz/DeclarationGenerator.java]: dibe

**Pull request merged**